main/src/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -136,7 +138,7 @@ public boolean isCanceled() {
             JSONUtilities.safePut(config, "state", "error");
             JSONUtilities.safePut(config, "error", "Error uploading data");
             JSONUtilities.safePut(config, "errorDetails", e.getLocalizedMessage());
-            return;
+            throw new IOException(e.getMessage());
```

Comment:
I have not checked thoroughly, but it seems curious that we need to throw an exception again after handling it above. Is it a problem with the architecture of the current code?

Reply:
To display on the UI we need to throw an exception here, we can not catch anything as an alternative. However, in general I think this file needs a major refactor but that would be for another ticket.

Reply:
Yes the state of this file is pretty terrible. Thanks for confirming.

Reply:
After looking more, I think this is fine for now and the only way to handle without changing the logic - an issue can be opened about cleaning up this file